lighthouse-docker/README.md

```diff
@@ -0,0 +1,177 @@
+# Lighthouse in Docker
+
+[![Docker Pulls](https://img.shields.io/docker/pulls/GoogleChrome/lighthouse.svg)](https://hub.docker.com/r/GoogleChrome/lighthouse/)
+
+> Run Lighthouse in a Docker container (as a CLI or a web service)
+
+This folder contains example `Dockerfile`s for running Lighthouse using
+[Headless Chrome](https://developers.google.com/web/updates/2017/04/headless-chrome)
+and full Chrome. It can be used in cloud environments like [Google App Engine Flex](https://cloud.google.com/appengine/docs/flexible/nodejs/) (Node), AWS, etc.
+
+## Usage
+
+Pull the pre-built image from [Dockerhub](https://hub.docker.com/):
+
+```bash
+docker pull GoogleChrome/lighthouse
+```
+
+> **Image size: ~805MB.**
+
+There are two ways to run the container:
+
+1. As a CLI.
+- As a REST API web service.
+
+### Run the CLI
+
+The container can be from the the CLI just like using the Lighthouse npm module. See
+Lighthouse docs for [CLI options](../#cli-options).
+
+```bash
+# Audit example.com. Lighthouse results are printed to stdout.
+docker run -it --rm --cap-add=SYS_ADMIN GoogleChrome/lighthouse https://example.com
+
+# Audits example.com and saves HTML report to a file.
+docker run -it --rm --cap-add=SYS_ADMIN GoogleChrome/lighthouse --quiet https://example.com > report.html
+
+# Audits example.com and saves JSON results to a file.
+docker run -it --rm --cap-add=SYS_ADMIN GoogleChrome/lighthouse --quiet --output=json https://example.com > report.json
+
+# Print the Lighthouse version.
+docker run -it --rm --cap-add=SYS_ADMIN GoogleChrome/lighthouse --version
+
+# Print Lighthouse CLI help.
+docker run -it --rm --cap-add=SYS_ADMIN GoogleChrome/lighthouse --help
+```
+
+### Run the REST API web service
+
+The container also bundles a web server that supports a REST API. You can
+use it to run Lighthouse return scores...in the cloud!
+
+To run the web server, invoke `docker run` without any arguments (e.g. no `CMD`):
+
+```bash
+docker run -dit -p 8080:8080 --rm --name GoogleChrome/lighthouse --cap-add=SYS_ADMIN GoogleChrome/lighthouse
+```
+
+This starts a server on `8080` and exposes a REST endpoint at
+`http://localhost:8080/audit`.
+
+**Examples**
+
+The endpoint supports `GET` requests with the following URL parameters:
+
+- `url`: parameter for the page you want to test.
+- `output`: report format. One of `json` or `html`.
+- `nolog`: By default, logs are streamed as Lighthouse audits the page.
+   Once the report is ready, the page redirects to the final report. If you
+   don't want this feature or your hosting env doesn't support [Server
+   Sent Events](https://www.html5rocks.com/en/tutorials/eventsource/basics/),
+   include the `nolog` parameter.
+
+```bash
+http://localhost:8080/audit?url=https://example.com&output=html
```

Comment:
I tried this exact URL. I saw the LH status updates but at the end I got a 404 for the report.html. `Cannot GET /report.1523317001312.html`